File: Kernel Simulator/Shell/Prompts/Presets/ZIP/PowerLine3.vb

'    Kernel Simulator  Copyright (C) 2018-2022  Aptivi
'
'    This file is part of Kernel Simulator
'
'    Kernel Simulator is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Kernel Simulator is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Imports System.IO
Imports System.Text
Imports KS.Misc.ZipFile

Namespace Shell.Prompts.Presets.ZIP
    Public Class ZipPowerLine3Preset
        Inherits PromptPresetBase
        Implements IPromptPreset

        Public Overrides ReadOnly Property PresetName As String = "PowerLine3" Implements IPromptPreset.PresetName

        Public Overrides ReadOnly Property PresetPrompt As String Implements IPromptPreset.PresetPrompt
            Get
                Return PresetPromptBuilder()
            End Get
        End Property

        Public Overrides ReadOnly Property PresetShellType As ShellType = ShellType.ZIPShell Implements IPromptPreset.PresetShellType

        Friend Overrides Function PresetPromptBuilder() As String Implements IPromptPreset.PresetPromptBuilder
            'PowerLine glyphs
            Dim TransitionChar As Char = Convert.ToChar(&HE0B0)

            'PowerLine preset colors
            Dim UserNameShellColorSegmentForeground As New Color(255, 255, 85)
            Dim UserNameShellColorSegmentBackground As New Color(127, 127, 43)
            Dim HostNameShellColorSegmentForeground As New Color(0, 0, 0)
            Dim HostNameShellColorSegmentBackground As New Color(255, 255, 85)
            Dim CurrentDirectoryShellColorSegmentForeground As New Color(0, 0, 0)
            Dim CurrentDirectoryShellColorSegmentBackground As New Color(255, 255, 255)
            Dim LastTransitionForeground As New Color(255, 255, 255)

            'Builder
            Dim PresetStringBuilder As New StringBuilder

            'File name
            PresetStringBuilder.Append(UserNameShellColorSegmentForeground.VTSequenceForeground)
            PresetStringBuilder.Append(UserNameShellColorSegmentBackground.VTSequenceBackground)
            PresetStringBuilder.AppendFormat(" {0} ", Path.GetFileName(ZipShell_FileStream.Name))

            'Transition
            PresetStringBuilder.Append(UserNameShellColorSegmentBackground.VTSequenceForeground)
            PresetStringBuilder.Append(HostNameShellColorSegmentBackground.VTSequenceBackground)
            PresetStringBuilder.AppendFormat("{0}", TransitionChar)

            'Current archive directory
            PresetStringBuilder.Append(HostNameShellColorSegmentForeground.VTSequenceForeground)
            PresetStringBuilder.Append(HostNameShellColorSegmentBackground.VTSequenceBackground)
            PresetStringBuilder.AppendFormat(" {0} ", ZipShell_CurrentArchiveDirectory)

            'Transition
            PresetStringBuilder.Append(HostNameShellColorSegmentBackground.VTSequenceForeground)
            PresetStringBuilder.Append(BackgroundColor.VTSequenceBackground)
            PresetStringBuilder.AppendFormat("{0} ", TransitionChar)

            'Present final string
            Return PresetStringBuilder.ToString()
        End Function

    End Class
End Namespace

File: Kernel Simulator/Shell/Prompts/Presets/RSS/PowerLine3.cs
﻿using System;

// Kernel Simulator  Copyright (C) 2018-2022  Aptivi
// 
// This file is part of Kernel Simulator
// 
// Kernel Simulator is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// Kernel Simulator is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.

using System.Text;
using ColorSeq;
using KS.ConsoleBase.Colors;
using KS.Kernel;
using KS.Misc.Text;
using KS.Shell.ShellBase.Shells;
using KS.Shell.Shells.RSS;

namespace KS.Shell.Prompts.Presets.RSS
{
    public class RSSPowerLine3Preset : PromptPresetBase, IPromptPreset
    {

        public override string PresetName { get; } = "PowerLine3";

        public override string PresetPrompt
        {
            get
            {
                return PresetPromptBuilder();
            }
        }

        public override ShellType PresetShellType { get; } = ShellType.RSSShell;

        internal override string PresetPromptBuilder()
        {
            // PowerLine glyphs
            char TransitionChar = Convert.ToChar(0xE0B0);

            // PowerLine preset colors
            var FirstColorSegmentForeground = new Color(255, 255, 85);
            var FirstColorSegmentBackground = new Color(127, 127, 43);
            var LastTransitionForeground = new Color(255, 255, 255);

            // Builder
            var PresetStringBuilder = new StringBuilder();

            // File name
            PresetStringBuilder.Append(FirstColorSegmentForeground.VTSequenceForeground);
            PresetStringBuilder.Append(FirstColorSegmentBackground.VTSequenceBackground);
            PresetStringBuilder.AppendFormat(" {0} ", new Uri(RSSShellCommon.RSSFeedLink).Host);

            // Transition
            PresetStringBuilder.Append(FirstColorSegmentBackground.VTSequenceForeground);
            PresetStringBuilder.Append(Flags.SetBackground ? ColorTools.BackgroundColor.VTSequenceBackground : Convert.ToString(CharManager.GetEsc()) + $"[49m");
            PresetStringBuilder.AppendFormat("{0} ", TransitionChar);

            // Present final string
            return PresetStringBuilder.ToString();
        }

        string IPromptPreset.PresetPromptBuilder() => PresetPromptBuilder();

    }
}